source/darwin/darwin_pki_utils.c

```diff
@@ -105,9 +105,40 @@ int aws_import_public_and_private_keys_to_identity(
     }
 
     if (key_status != errSecSuccess && key_status != errSecDuplicateItem) {
-        AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: error importing private key with OSStatus %d", (int)key_status);
-        result = aws_raise_error(AWS_IO_FILE_VALIDATION_FAILURE);
-        goto done;
+        if (key_status == errSecUnknownFormat) { // If the format is unknow, try ecc
+            AWS_LOGF_TRACE(AWS_LS_IO_PKI, "static: error reading private key format, try ECC key format.");
+            struct aws_byte_buf der_buffer;
+            AWS_ZERO_STRUCT(der_buffer);
```

Comment:
needed? init function should always zero out the buffer

Reply:
The function is deprecated.